### PR TITLE
[Choreo] K8s artefacts for rate-limiter service

### PR DESCRIPTION
**Pull request by slahirucd7**
### Purpose
Adds K8 artefacts to work with the rate-limiter service.

image

### Issues

Fixes #

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: resources/k8s-artifacts/choreo-connect/rate-limiter-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# --------------------------------------------------------------------
+# Copyright (c) 2023, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (renuka-fernando) [inline resources/k8s-artifacts/choreo-connect/rate-limiter-deployment.yaml:2]:
```suggestion
# Copyright (c) 2023, WSO2 LLC. (http://wso2.com) All Rights Reserved.
```

File: resources/k8s-artifacts/choreo-connect/rate-limiter-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------
+
+# Deployment for rate-limiter
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: rate-limiter-deployment
```
Comment (renuka-fernando) [inline resources/k8s-artifacts/choreo-connect/rate-limiter-deployment.yaml:22]: Shall we name this "choreo-connect-rate-limiter" to make it consistent with other components

File: resources/k8s-artifacts/choreo-connect/rate-limiter-service.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: rate-limiter-deployment
+  name: rate-limiter
+spec:
+  type: ClusterIP
+  selector:
+    app: rate-limiter-deployment
```
Comment (renuka-fernando) [inline resources/k8s-artifacts/choreo-connect/rate-limiter-service.yaml:28]: choreo-connect-rate-limiter

File: resources/k8s-artifacts/choreo-connect/rate-limiter-service.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# service for rate-limiter
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: choreo-connect-rate-limiter
+  name: rate-limiter
+spec:
+  type: ClusterIP
```
Comment (renuka-fernando) [inline resources/k8s-artifacts/choreo-connect/rate-limiter-service.yaml:26]:
```suggestion
  type: ClusterIP
  clusterIP: None
```
Let's use a headless service

Comment (slahirucd7) [reply resources/k8s-artifacts/choreo-connect/rate-limiter-service.yaml:26]: fixed with: https://github.com/wso2/product-microgateway/pull/3281/commits/a20b1f2b377af726154bfa5eba80694b2ac7bfa6

**Review by renuka-fernando — APPROVED** [review]

**Review by pubudu538 — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : dev-deployment-v2 : 20230309.6

**Comment by choreo-cicd** [comment]
[failed] DataplaneE1 cluster : dev-deployment-v2 : 20230309.6

**Comment by choreo-cicd** [comment]
[failed]  : dev-deployment-v2 : 20230309.6

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20230309.27

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230310.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230310.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CentralIN) cluster : stage-deployment-v2 : 20230310.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230310.1

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : prod-deployment-v2 : 20230310.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CentralIN) cluster : prod-deployment-v2 : 20230310.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230310.3

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230310.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CentralIN) cluster : prod-deployment-v2 : 20230310.3

**Pull request merged**